10/README.md

```diff
@@ -4,29 +4,70 @@ Name: A Strangler Application Approach to Rewriting Some Code in Go
+BigchainDB 2.0 Alpha 1 is now the composition of three different system (MongoDB, Tendermint, and BigchainDB itself).
+
+This setup generates different problems:
+
+1. managing 3 services together adds operational overhead and complicates distribution;
+
+    Blockchain clients are commonly independent binaries. This is true for Bitcoin, Ethereum, and Monero.
+
+2. the current design raises certain performance concerns;
+
+    The TCP communications between BigchainDB and Tendermint, the transaction validation implemented in Python, and Crypto-Conditions implemented in Python might significantly contribute to the performance degradation. The detailed benchmarks SHOULD follow.
+
+3. there is data duplication between LevelDB and MongoDB;
+
+    Currently, the only way to replace the database backend in Tendermint is to implement a Tendermint-based service in Go, with the database interface implemented according to the needs.
+
+4. we need to maintain [`py-abci`][pyabci], which is not an official implementation;
+
+5. the configuration is spread across BigchainDB and Tendermint.
+
+    A way to overcome the issue is to implement a Tendermint-based service in Go, with the overwritten config loading/parsing mechanisms.
+
+The BEP does not aim to solve all of the problems described above but suggests a reasonable small experimental first step towards the new design.
+
+## Problem breakdown
+
+* The BEP suggests to replace the remote ABCI proxy app with the HTTP server that serves `POST /api/v1/validate`.
+* The BEP suggests to produce a custom binary based on Tendermint where
+  * the Golang ABCI interface implementation either directly executes ABCI commands or proxies them to the HTTP server;
+  * the HTTP server proxies `POST /api/v1/transactions` to the corresponding endpoint in BigchainDB Server.
 
 ## Overview
-In this BEP, we evaluate a rewrite of some parts of BigchainDB in another language, specifically Golang. The benefits are multiple: Golang it is a fairly simple language, easy to learn and read; it is shown to be efficient and faster than Python and can compile to a binary file; it is used in several big projects, and it's the language of choice for major blockchain projects: this simplifies code reuse if we need it in the future, and allows us a tighter integration with Tendermint.
+In this BEP, we evaluate a rewrite of some parts of BigchainDB in Golang.
+
+The main reason to choose Golang is:
+
+* Tendermint is written in Golang. Using Go allows to achieve the most efficient integration with Tendermint. It includes efficient ABCI communications, interactions with the database, and straightforward compilation of BigchainDB and Tendermint into a single binary.
+
+
+Other notes about Go:
```

Comment:
I would like to understand the reason behind writing these notes. IMHO the document should only have reason for leaning towards Golang in comparison to python. So, I think these notes should be contrasted with python.